src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.model.patient.comparators.NameComparator.NAME_COMPARATOR;
+import static seedu.address.model.patient.comparators.PreferredNameComparator.PREFERRED_NAME_COMPARATOR;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.Model;
+import seedu.address.model.patient.Patient;
+
+/**
+ * Sorts (based on specified comparator) and list all patients in the address book to the user.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_SORT_SUCCESS = "Successfully sorted all patients by ";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sort all persons by the specified attribute "
+            + "and displays them as a list with index numbers.\n"
+            + "Parameters: [ATTRIBUTE]\n"
+            + "Example: " + COMMAND_WORD + " n";
+
+    private static final Logger logger = LogsCenter.getLogger(SortCommand.class);
+
+    private final Comparator<Patient> comparator;
+    private final String sortAttribute;
+
+    /**
+     * Creates a new SortCommand to sort the patients in patient list by the specified {@code comparator}
+     *
+     * @param comparator the comparator used to compare the patient list
+     */
+    public SortCommand(Comparator<Patient> comparator, String sortAttribute) {
+        requireAllNonNull(comparator, sortAttribute);
+
+        this.comparator = comparator;
+        this.sortAttribute = sortAttribute;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        logger.log(Level.INFO, "Attempting to execute SortCommand");
+
+        requireNonNull(model);
+
+        if (this.comparator.equals(NAME_COMPARATOR)) {
+            assert (sortAttribute.equals("name")) : "Sort Attribute and Comparator mismatch!";
+        }
+
+        if (this.comparator.equals(PREFERRED_NAME_COMPARATOR)) {
+            assert (sortAttribute.equals("preferred name")) : "Sort Attribute and Comparator mismatch!";
+        }
+
+        List<Patient> patientList = model.getFullPatientList();
+        List<Patient> patientArrayList = new ArrayList<>(patientList);
+
+        Collections.sort(patientArrayList, this.comparator);
+
+        model.updatePatientList(patientArrayList);
+        model.updateFilteredPatientList(Model.PREDICATE_SHOW_ALL_PATIENTS);
+
+        logger.log(Level.INFO, "Successfully sorted the patient list by " + this.sortAttribute);
+
+        return new CommandResult(MESSAGE_SORT_SUCCESS + sortAttribute);
+    }
```

Comment:
Considering that the Comparator and sortAttribute are inherently linked, is there a necessity to pass both and then validate through assertions by comparing them? The input is determined by the parser logic. Instead, it might be simpler to pass in just one, and then create the corresponding Comparator or sortAttribute accordingly. This approach could potentially reduce the likelihood of errors occurring.

Reply:
Good suggestion! That was my initial design for the code where I would get the sortAttribute based on checks on the Comparator, however I felt that since I am parsing the sort attribute in the sortCommandParser, I could get the sort attribute there, so I decided to change it.

But now that I revisit this code, I think it would be better to create the corresponding sortAttribute to reduce the likelihood of errors occurring! Will proceed to change it!